conman/routes/managers.py

```diff
@@ -30,3 +32,31 @@ def best_match_for_path(self, path):
         except IndexError:
             msg = 'No matching Route for URL. (Have you made a root Route?)'
             raise self.model.DoesNotExist(msg)
+
+    def move_branch(self, old_url, new_url):
+        """
+        Move this Route and all descendants to a new URL.
+
+        Movement will be atomic. This means that failure to move any one Route
+        will cause all movement to fail. A conflicting URL will cause an
+        IntegrityError.
+        """
+        to_move = self.filter(url__startswith=old_url)
+        to_move.update(url=Concat(
+            Value(new_url),
+            Substr('url', len(old_url) + 1),  # 1 indexed
+        ))
+
+    def swap_urls(self, path_a, path_b):
+        """
+        Swap the urls of two Routes without moving their descendants.
+
+        If DEFERRED unique contrstraints get supported in django, we might
+        benefit from revisiting how this works (ie: dropping the UUID).
+
+        See https://code.djangoproject.com/ticket/20581.
+        """
+        tmp_path = str(uuid.uuid4())
+        self.filter(url=path_a).update(url=tmp_path)
+        self.filter(url=path_b).update(url=path_a)
+        self.filter(url=tmp_path).update(url=path_b)
```

Comment:
Would it be better to do a `get`, and raise errors when `Routes` don't exist? If so, should we be doing this in an atomic transaction?

Would [`select_for_update`](https://docs.djangoproject.com/en/1.9/ref/models/querysets/#select-for-update) be worth considering to avoid race conditions? Is that a threat?